### DISCUSSION TITLE
How to install node.js in latest beta?

### DISCUSSION THREAD

**Discussion by AlejandroAkbal** [serversideup/docker-php#325] [Q&A]
I want a specific node version, not just the latest available

**Comment by jaydrogers** [comment]
Installing node is outside the scope of our images.

We always recommend separating the two like this example 

https://github.com/serversideup/financial-freedom/blob/b528f0853765a1f75d2154ecee823bb56ee82d7a/docker-compose.dev.yml#L35-L77

**Reply by AlejandroAkbal** [reply]
Does that example use Node.js for building and SSR?

What are the benefits of separating them?

**Reply by jaydrogers** [reply]
You can go to the repo and open it up in GitPod. Everything will come up for you automatically: https://github.com/serversideup/financial-freedom/

![CleanShot 2024-04-25 at 08 08 05@2x](https://github.com/serversideup/docker-php/assets/3174134/b3ac83e4-9d01-4b09-b2bc-064324eab88c)

### Building with Node
You only need Node at build time, so there really isnt a reason to put it in your final image.

### Node Based SSR
Since it is a Node server, that should be separated

### Why do we do that?
We believe a Docker container should do one thing very well. Separating it out can make it easier for you to maintain, test, and ensure your final image is efficient with the smallest security footprint possible.

### Laravel API + Nuxt Example
We did a Laravel API + Nuxt SSR Web App + Nuxt SPA Mobile app in the book. It doesn't cover deployments in the book, but The Complete Package does include source code on how we ship it to production https://serversideup.net/ultimate-guide-to-building-apis-and-spas-with-laravel-and-nuxt3/
